python/pyarrow/dataset.py

```diff
@@ -731,6 +731,24 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         (e.g. S3)
     max_partitions : int, default 1024
         Maximum number of partitions any batch may be written into.
+    file_visitor : Function
+        If set, this function will be called with a WrittenFile instance
+        for each file created during the call.  This object will have both
+        a path attribute and a metadata attribute.
+
+        The path attribute will a str containing the absolute path to
```

Comment:
Based on a quick test, it are not always absolute paths, but it depends on what has been passed as base_dir for the written dataset (whether that's an absolute or relative path).